src/v/cluster/node_status_backend.cc

```diff
@@ -244,13 +260,24 @@ result<node_status> node_status_backend::process_reply(
   model::node_id target_node_id, result<node_status_reply> reply) {
     vassert(ss::this_shard_id() == shard, "invoked on a wrong shard");
     static constexpr auto rate_limit = std::chrono::seconds(1);
+
+    auto reset_timeout_counter = ss::defer([this, target_node_id]() {
+        this->_discovered_peers.at(target_node_id).consecutive_timeouts = 0;
+    });
+
     if (reply.has_error()) {
         auto err = reply.error();
         if (
           err.category() == rpc::error_category()
           && static_cast<rpc::errc>(err.value())
                == rpc::errc::client_request_timeout) {
             _stats.rpcs_timed_out += 1;
+
+            // register consecutive timeouts
+            reset_timeout_counter.cancel();
```

Comment:
wondering if the counter should be reset to 0 only on one successful request (which ensures the transport queue is drained).. this block means if there is a different error other than request_timeout also resets the counter IIUC... wdyt